tools/packaging/static-build/cache_components.sh

```diff
@@ -11,34 +11,101 @@ set -o pipefail
 script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 source "${script_dir}/../scripts/lib.sh"
-source "${script_dir}/qemu/build-static-qemu-cc.sh"
 
 export KATA_BUILD_CC="${KATA_BUILD_CC:-}"
 export qemu_cc_tarball_name="kata-static-qemu-cc.tar.gz"
 
 cache_qemu_artifacts() {
+	source "${script_dir}/qemu/build-static-qemu-cc.sh"
 	local current_qemu_version=$(get_from_kata_deps "assets.hypervisor.qemu.version")
-	create_qemu_cache_asset "${qemu_cc_tarball_name}" "${current_qemu_version}"
+	create_cache_asset "${qemu_cc_tarball_name}" "${current_qemu_version}"
 	local qemu_sha=$(calc_qemu_files_sha256sum)
         echo "${current_qemu_version} ${qemu_sha}" > "latest"
 }
 
-create_qemu_cache_asset() {
+cache_clh_artifacts() {
+	local binary="cloud-hypervisor"
+	local binary_path="$(echo $script_dir | sed 's,/*[^/]\+/*$,,' | sed 's,/*[^/]\+/*$,,' | sed 's,/*[^/]\+/*$,,')"
+	echo "binary path $binary_path"
+	local current_cloud_hypervisor_version=$(get_from_kata_deps "assets.hypervisor.cloud_hypervisor.version")
+	local clh_binary_path="${binary_path}/tools/packaging/kata-deploy/local-build/build/cc-cloud-hypervisor/builddir/cloud-hypervisor"
+	if [ -f "${clh_binary_path}/cloud-hypervisor" ]; then
+		cp "${clh_binary_path}/${binary}" .
+	else
+		cloud_hypervisor_build_path="${binary_path}/cloud-hypervisor"
+		cp "${cloud_hypervisor_build_path}/${binary}" .
+	fi
+	create_cache_asset "${binary}" "${current_cloud_hypervisor_version}"
+	echo "${current_cloud_hypervisor_version}"  > "latest"
+}
+
+create_cache_asset() {
 	local component_name="$1"
 	local component_version="$2"
-	local qemu_cc_tarball_path=$(sudo find / -iname "${qemu_cc_tarball_name}")
-	info "qemu cc tarball_path ${qemu_cc_tarball_path}"
-	cp -a "${qemu_cc_tarball_path}" .
+	local verify_qemu=$(echo "${component_name}" | grep qemu || true)
+	local verify_clh=$(echo "${component_name}" | grep cloud || true)
+
+	if  [ ! -z "${verify_qemu}" ]; then
```

Comment:
Why don't you move this qemu specific block to `cache_qemu_artifacts()`? Then you don't need to set the `verify_qemu` and `verify_clh` variables.